gitbook-docs/security.md

```diff
@@ -0,0 +1,118 @@
+# Security
+
+## Communications Security
+
+For privacy and data integrity REST and WebSockets communications should be secured with Transport Layer Security
+(TLS). All communications over unsecure protocols like HTTP and WebSockets without TLS must be considered insecure even
+with authentication and access control mechanisms in place.
```

Comment:
Perhaps we should word this even more strongly: require TLS for authentication and disable authentication in the case where TLS is not available. Sending passwords in the clear is worse than not having them at all. People tend to reuse passwords and sending them in the clear could compromise their other accounts.

Reply:
Given how complicated cert management is and how bad the experience in browsers with self signed certs is I do not think that is realistic.

Reply:
What's not feasible about saying "you don't have a certificate on your server, you don't get to use authentication features until you do"? My point is that it is better to make everything wide open then to pretend that there is any security at all by providing a login mechanism over an unencrypted connection.

Reply:
If we make TLS a hard requirement I doubt we will make much progress with this as it will be prohibitively inconvenient for Joe Boater. 

I believe there is still value in authentication and authorization without TLS.

There are holes in SK TLS support that need resolving. We should fix those first adn try to make TLS adption as easy as possible first before making it a requirement.

Reply:
I’m not saying that it should be a requirement for implementing Signal K, just that it should be a requirement if you want to use authentication/authorization.

Reply:
I think we should just recommend TLS in the spec.

I agree with @tkurki there is still value in having authentication without TLS.